stdlib/public/Reflection/TypeLowering.cpp

```diff
@@ -1295,12 +1896,39 @@ class EnumTypeInfoBuilder {
         Alignment = FixedDescriptor->getAlignment();
         NumExtraInhabitants = FixedDescriptor->NumExtraInhabitants;
         BitwiseTakable = FixedDescriptor->isBitwiseTakable();
+        unsigned Stride = ((Size + Alignment - 1) & ~(Alignment - 1));
+        if (Stride == 0)
+          Stride = 1;
+
+        auto PayloadSize = EnumTypeInfo::getPayloadSizeForCases(Cases);
+        BitMask spareBitsMask(PayloadSize);
+        auto validSpareBitsMask = populateSpareBitsMask(Cases, spareBitsMask);
+
+        if (!validSpareBitsMask) {
```

Comment:
Here's the critical logic that determines how to handle a particular MPE based on our current understanding.